### ISSUE TITLE
500 Error in Query Builder with new MSSQL tables

### ISSUE THREAD

**Issue by sebcodes** [grafana/grafana#81935]
### What happened?

When I run the query builder:

`logger=tsdb.mssql endpoint=queryData pluginId=mssql dsName="grafanadb.local - (MSDB)" dsUID=a9a3b5b6-9bf3-4e13-b7bf-65174a56e0e1 uname=grafana-test t=2024-02-06T09:08:50.7282003+01:00 level=error msg="ExecuteQuery panic" error="runtime error: invalid memory address or nil pointer dereference" stack=[]`

Other "old" tables work perfectly, i think grafana can't create cache/infos about the db in the configuration db.
DB is running on postgres

### What did you expect to happen?

A succesful Query, the Query work on MSSQL

### Did this work before?

On 10.2.2 OSS

### How do we reproduce it?

1. Add MSSQL Connection
2. Try to run Query on a table with more than 100 rows or emtpy table
3.

### Environment (with versions)?

Grafana: 10.3.1 OSS
OS: Windows
Browser: Firefox

### Grafana platform?

A downloaded binary

### Datasource(s)?

mssql
```[tasklist]
### Tasks
```

**Comment by sebcodes** [comment]
Hi, 
it appears also by new datasources and same queries.

Also a fresh Installation with 10.3.1 give me this error.

**Comment by aangelisc** [comment]
Hi @sebcodes,

Apologies for the delayed response. I am unable to replicate this issue. Are you using the query builder or the code builder to execute the query? Have you also managed to replicate this on 10.4.0? Please let me know and I'll investigate further 😊

**Comment by sebcodes** [comment]
Yes, it happens also on v10.4.0 OSS by both builders.

**Comment by aangelisc** [comment]
Are you able to provide an example schema of the table you're attempting to query and possible the dashboard query as well? Does this work if Grafana is not configured with a Postgres database?

**Comment by sebcodes** [comment]
Of course. Same Problem with the embedded sqllite db.
Table:
![grafik](https://github.com/grafana/grafana/assets/52137481/a20e4599-9b59-4cf5-93fd-8100c23ded86)
Grafana Error in both builders:
![grafik](https://github.com/grafana/grafana/assets/52137481/5c58f3c2-b727-4a28-adbd-ed4f8bd1540f)
![grafik](https://github.com/grafana/grafana/assets/52137481/8b392cd2-0ad9-4e24-b96f-4f0ceb4a2d91)

**Comment by aangelisc** [comment]
Hi @sebcodes,

I am still unable to replicate this. I've tested using 10.4.0 on both Windows and MacOS. 

- Does this error happen with any SQL Server database you try to query?  
- Are you able to enable debug logging and share the logs (see [here](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#log) for info)
- Do you see this error if you run Grafana in a Docker container?
- Are you using the default Grafana configuration or have you modified the `ini`?
- Is your MSSQL data source configured with the default settings or have you modified anything else?

**Comment by sebcodes** [comment]
**Does this error happen with any SQL Server database you try to query?**
No, i created mutiple tables on different databases, only new tables throw this error, all other dashboards that use the same endpoint working fine.
**Are you able to enable debug logging and share the logs ?**
`logger=tsdb.mssql endpoint=queryData pluginId=mssql dsName="srv01.local - Monitoring" dsUID=ea4b46f2-84ce-4bba-883a-c1aff3116e33 uname=A-Kiefer t=2024-03-20T07:13:59.8646986+01:00 level=error msg="ExecuteQuery panic" error="runtime error: invalid memory address or nil pointer dereference" stack=[]
logger=context userId=20 orgId=1 uname=Admin t=2024-03-20T07:13:59.8652367+01:00 level==downstreaminfo msg="Request Completed" method=POST path=/api/ds/query status=400 remote_addr=10.56.128.192 time_ms=25 duration=25.7213ms size=125 handler=/api/ds/query status_source`

**Do you see this error if you run Grafana in a Docker container?**
Yes
**Are you using the default Grafana configuration or have you modified the ini?**
I tried both, same behavior
**Is your MSSQL data source configured with the default settings or have you modified anything else?**
Default Settings, i tried different settings

**Comment by aangelisc** [comment]
Hi @sebcodes,

Apologies for the delay. In attempting to reproduce and debug your issue we noticed that the error logging for MSSQL on Windows was not working as expected. This has been rectified by #85118. If possible tomorrow, would you be able to make use of the nightly Grafana build (available [here](https://grafana.com/grafana/download/11.1.0-169129?pg=get&plcmt=selfmanaged-box1-cta1&edition=oss)) to demonstrate this issue again and supply us with the logs. I'm hopeful that this will reveal the true stack trace and allow us to identify the source of the issue.

Thanks!

**Comment by sebcodes** [comment]
Hello, of course I can try that, but only from next Wednesday, until then I don't have the opportunity

**Comment by aangelisc** [comment]
That's perfect! Thank you and I look forward to hearing back 😊

**Comment by sebcodes** [comment]
Good morning, 
the new build is installed but i got the same error message as before.

`logger=tsdb.mssql endpoint=queryData pluginId=mssql dsName="srv01.local - Monitoring" dsUID=ea4b46f2-84ce-4bba-883a-c1aff3116e33 uname=admin t=2024-04-03T07:47:23.2456997+02:00 level=error msg="ExecuteQuery panic" error="runtime error: invalid memory address or nil pointer dereference" stack=[]
logger=context userId=20 orgId=1 uname=admin t=2024-04-03T07:47:23.2462351+02:00 level=info msg="Request Completed" method=POST path=/api/ds/query status=400 remote_addr=10.56.128.192 time_ms=8 duration=8.5904ms size=125 referer="https://srv01.local:3000/explore?orgId=1&panes=%7B%221lc%22%3A%7B%22datasource%22%3A%22ea4b46f2-84ce-4bba-883a-c1aff3116e33%22%2C%22queries%22%3A%5B%7B%22refId%22%3A%22A%22%2C%22datasource%22%3A%7B%22type%22%3A%22mssql%22%2C%22uid%22%3A%22ea4b46f2-84ce-4bba-883a-c1aff3116e33%22%7D%2C%22format%22%3A%22table%22%2C%22rawSql%22%3A%22%22%2C%22editorMode%22%3A%22builder%22%2C%22dataset%22%3A%22DBMonitoring%22%7D%5D%2C%22range%22%3A%7B%22from%22%3A%22now-1h%22%2C%22to%22%3A%22now%22%7D%7D%7D&schemaVersion=1" handler=/api/ds/query status_source=downstream`

**Comment by sebcodes** [comment]
It's 11.1.0-169129
I'm using Windows Server 2022
Perhabs i can try it on Rhel9 but it must work on the windows machine. 
Tomorrow i analyse the database table configuration again to exclude the database as problem source

**Comment by aangelisc** [comment]
Thanks, I'd appreciate if you can try on RHEL so that we can get a better idea of if it's a data source/Grafana issue or something OS specific.

**Comment by sebcodes** [comment]
I think i found the problem.
My column data type was "sql_variant", if i changed it to "nchar" it works completely fine. Could you test it with the "sql_variant" datatype? Or maybe Grafana would inplement the use of "sql_variant"

**Comment by aangelisc** [comment]
Thank you for that additional information @sebcodes! I was able to replicate and I've created two PR's as an outcome of this issue. The #85823 will add support for columns of type `sql_variant` to the data source. I've tested this locally and it seems to work as expected.

grafana/grafana-plugin-sdk-go#951 updates the error messaging from the SDK to provide the user with insight as to which column of theirs may be using an unsupported data type.
